planning/obstacle_velocity_limiter/src/obstacle_velocity_limiter_node.cpp

```diff
@@ -245,6 +266,35 @@ bool ObstacleVelocityLimiterNode::validInputs(const boost::optional<size_t> & eg
 
   return occupancy_grid_ptr_ && dynamic_obstacles_ptr_ && ego_idx && current_odometry_ptr_;
 }
+
+void ObstacleVelocityLimiterNode::onLanelet2Map(autoware_auto_mapping_msgs::msg::HADMapBin::SharedPtr msg){
+
+    lanelet::utils::conversion::fromBinMsg(*msg, lanelet_map_ptr_);
+
+    if (msg->header.frame_id == "map"){
+        extractStaticObstacles(*lanelet_map_ptr_, obstacle_params_.static_map_tags);
+        return;
+    }
+}
+
+void ObstacleVelocityLimiterNode::transformInputsToOdomFrame() {
+    geometry_msgs::msg::TransformStamped transform_stamped =
+            map_odom_tf_buffer_->lookupTransform("odom", "map", tf2::TimePointZero);
+
+    // transform occupancy grid
+    occupancy_grid_ptr_->header.frame_id = "odom";
+    occupancy_grid_ptr_->header.stamp = this->now();
+    tf2::doTransform(occupancy_grid_ptr_->info.origin,
+                     occupancy_grid_ptr_->info.origin, transform_stamped);
+    occupancy_grid_pub_->publish(*occupancy_grid_ptr_);
+
+    // transform dynamic obstacles
+    dynamic_obstacles_ptr_->header.frame_id = "odom";
+    dynamic_obstacles_ptr_->header.stamp = this->now();
+    dynamic_obstacles_pub_->publish(*dynamic_obstacles_ptr_);
```

Comment:
This doesn't actually apply any transform...

Reply:
Yes. This is my to-do list. I haven't dived deeper into dynamic_obstacles yet. If the predicted_objects (aka dynamic obstacles) are already in odom frame. Then the transform is not needed.

Reply:
They will be in `odom` frame, yes.